app/controllers/dashboard_controller.rb

```diff
@@ -10,6 +10,45 @@ def index
     end
   end
 
+  def create
+    Dashboard::Manager.add_widget_to_user(User.current, params[:widget])
+    redirect_to root_path
+  end
+
+  def destroy
+    widget = Widget.find(params[:id])
```

Comment:
Use ```find_resource``` which will put the appropriate permissions scope. You'll have to override ```resource_class``` though.